src/components/checkbox/checkbox.test.js

```diff
@@ -0,0 +1,549 @@
+/* eslint-disable no-shadow */
+import * as React from "react";
+import Checkbox from "./checkbox.component";
+import CheckboxGroup from "./checkbox-group.component";
+import {
+  checkboxComponent,
+  checkboxRole,
+  checkboxGroup,
+  checkboxLabel,
+  checkboxgroupLegend,
+  checkboxInlineFieldHelp,
+  checkboxIcon,
+  checkboxGroupIcon,
+  checkboxSvg,
+  checkboxHelpIcon,
+} from "../../../cypress/locators/checkbox";
+import {
+  fieldHelpPreview,
+  getComponent,
+  tooltipPreview,
+} from "../../../cypress/locators/index";
+import CypressMountWithProviders from "../../../cypress/support/component-helper/cypress-mount";
+import {
+  verifyRequiredAsteriskForLegend,
+  verifyRequiredAsteriskForLabel,
+} from "../../../cypress/support/component-helper/common-steps";
+
+const specialCharacters = ["mp150ú¿¡üßä", "!@#$%^*()_+-=~[];:.,?{}&\"'<>"];
+const testData = "cypress_data";
+
+const CheckboxComponent = ({ ...props }) => {
+  const [setIsChecked] = React.useState(false);
+  return (
+    <>
+      <div
+        style={{
+          marginTop: "64px",
+        }}
+      >
+        <Checkbox
+          label="Checkbox 1"
+          checked={() => setIsChecked(false)}
+          onChange={(e) => setIsChecked(e.target.checked)}
+          {...props}
+        />
+      </div>
+    </>
+  );
+};
+
+const CheckboxGroupComponent = ({ children, ...props }) => {
+  const checkboxGroupName = "checkboxGroupName";
+  const [setIsChecked] = React.useState(false);
+  return (
+    <div
+      style={{
+        marginTop: "64px",
+        marginLeft: "64px",
+      }}
+    >
+      <CheckboxGroup
+        id="checkboxgroup"
+        name="checkboxgroup"
+        groupName="{checkboxGroupName} "
+        legend="Test CheckboxGroup Label"
+        {...props}
+      >
+        <Checkbox
+          label="Required"
+          id={`checkbox_${checkboxGroupName}-Required`}
+          key={`checkbox_${checkboxGroupName}-Required`}
+          name={`checkbox_${checkboxGroupName}-Required`}
+          checked={() => setIsChecked(false)}
+          onChange={(e) => setIsChecked(e.target.checked)}
+        />
+        <Checkbox
+          label="Optional"
+          id={`checkbox_${checkboxGroupName}-Optional`}
+          key={`checkbox_${checkboxGroupName}-Optional`}
+          name={`checkbox_${checkboxGroupName}-Optional`}
+          checked={() => setIsChecked(false)}
+          onChange={(e) => setIsChecked(e.target.checked)}
+        />
+
+        {children}
+      </CheckboxGroup>
+    </div>
+  );
+};
+
+context("Testing Checkbox component", () => {
+  describe("should render Checkbox component", () => {
+    it("should render Checkbox component with data-component", () => {
+      CypressMountWithProviders(
+        <CheckboxComponent data-component={testData} />
+      );
+      getComponent(testData).should("have.attr", "data-component", testData);
```

Comment:
we still need to start from `checkboxComponent()` and then only `.should()`

Reply:
Hi Igor, checkboxComponent() won't work here as it requires the data component to be set to "checkbox". In this test I've mounted the checkbox setting the component to "cypress_data".